File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletUVLightV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletUVLightV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your UV Light Bricklet 2.0

// Callback function for UV index callback
function cb_uvi($uvi)
{
    echo "UV Index: " . $uvi/10.0 . "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$uvl = new BrickletUVLightV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register UV index callback to function cb_uvi
$uvl->registerCallback(BrickletUVLightV2::CALLBACK_UVI, 'cb_uvi');

// Set period for UVI callback to 1s (1000ms) without a threshold
$uvl->setUVICallbackConfiguration(1000, FALSE, 'x', 0, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/go/example_callback.go
package main

import (
	"fmt"
	"tinkerforge/ipconnection"
	"tinkerforge/uv_light_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your UV Light Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	uvl, _ := uv_light_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	uvl.RegisterUVICallback(func(uvi int32) {
		fmt.Printf("UV Index: %d\n", float64(uvi)/10.0)
	})

	// Set period for UVI callback to 1s (1000ms) without a threshold.
	uvl.SetUVICallbackConfiguration(1000, false, 'x', 0, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
